sgv2-graphqlapi/src/test/java/io/stargate/sgv2/graphql/schema/GraphqlTestBase.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.sgv2.graphql.schema;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.quality.Strictness.LENIENT;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import graphql.ExecutionInput;
+import graphql.ExecutionResult;
+import graphql.GraphQL;
+import graphql.GraphQLError;
+import graphql.execution.AsyncExecutionStrategy;
+import graphql.schema.GraphQLSchema;
+import io.stargate.proto.QueryOuterClass.Batch;
+import io.stargate.proto.QueryOuterClass.Query;
+import io.stargate.proto.QueryOuterClass.QueryParameters;
+import io.stargate.proto.QueryOuterClass.Response;
+import io.stargate.proto.QueryOuterClass.ResultSet;
+import io.stargate.proto.QueryOuterClass.Value;
+import io.stargate.proto.Schema.CqlKeyspaceDescribe;
+import io.stargate.sgv2.common.grpc.StargateBridgeClient;
+import io.stargate.sgv2.graphql.web.resources.StargateGraphqlContext;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = LENIENT)
```

Comment:
While I understand the intention behind Mockito strictness, I find it really annoying in cases like this. This class sets up a generic infrastructure that gets reused by dozens of tests, in many different ways: the GraphQL query may not even parse, the fetcher might throw, or it might work and execute a CQL query, or a batch, that we may or may not want to mock the results thereof. So invariably not all mocks will be invoked in every case.
If we opt for `lenient()`, I think it will be needed for almost every `when()` call, so it's not much better than annotating the class.
The other approach (and I think what Mockito advocates) is to have each test mock precisely what it needs, in other words avoid this parent class. I think that's not practical.

Reply:
Yes but annotating the whole class delegates behavior to the sub-classes.. My intention was that setup in supper class is `lenient` (by using `lenient().when(...)`) and then sub-classes have to use extra mocking in the correct way.. Anyway not sure how much of mocking is there anyway in the sub-classes..